src/plugins/discover/public/application/apps/main/components/layout/discover_layout.tsx

```diff
@@ -69,6 +72,17 @@ export function DiscoverLayout({
 
   const [expandedDoc, setExpandedDoc] = useState<ElasticSearchHit | undefined>(undefined);
   const [inspectorSession, setInspectorSession] = useState<InspectorSession | undefined>(undefined);
+  const [discoverViewMode, setDiscoverViewMode] = useState(DISCOVER_VIEW_MODES.DOCUMENT_LEVEL);
```

Comment:
:art: Leaving aside my concern around having this configurable at all (see other comment), `uiSettings` should not really change as a reference (i.e. the service itself never changes), thus there should be no real use-case for setting this value via a `useEffect`below and causing one initial render more, but just retrieve the default value in the `useState` initial value?

```tsx
  const [discoverViewMode, setDiscoverViewMode] = useState(() => uiSettings?.get(AGGREGATED_VIEW_SETTING) ? DISCOVER_VIEW_MODES.FIELD_LEVEL : DISCOVER_VIEW_MODES.DOCUMENT_LEVEL);
```

Reply:
Good call, but I'll resolve this for now as I have moved this setting to Discover app state instead of advanced settings.